### PR TITLE
feat(builtgateway): document builtin gateway and policies

### PR DESCRIPTION
**Pull request by lahabana**
Signed-off-by: Charly Molter <charly.molter@konghq.com>

### CODE REVIEW

File: docs/docs/dev/documentation/gateway.md [hunk 1]
```diff
@@ -1,15 +1,35 @@
 # Gateway
 
-The `Dataplane` entity can operate in `gateway` mode. This way you can integrate Kuma with existing API Gateways like [Kong](https://github.com/Kong/kong).
+Kuma Gateway is a Kuma component that routes network traffic from outside a Kuma mesh to services inside the mesh.
+The gateway can be thought of as having 1 foot outside the mesh to receive traffic and one foot inside the mesh to route this external traffic to services inside the mesh.
```
Comment (michaelbeaumont) [inline docs/docs/dev/documentation/gateway.md:4]:
```suggestion
The gateway can be thought of as having one foot outside the mesh to receive traffic and one foot inside the mesh to route this external traffic to services inside the mesh.
```

File: docs/docs/dev/documentation/gateway.md [hunk 2]
```diff
@@ -1,15 +1,35 @@
...
-Security for an entrance to the mesh should be handled by API Gateway itself.
+Security for an entrance to the mesh should be handled by Gateway itself.
+
+Kuma Gateway is deployed as a Kuma Dataplane, i.e. an instance of the `kuma-dp` process.
+Like all Kuma Dataplanes, the Kuma Gateway Dataplane manages a co-located Envoy proxy process that does the actual network traffic proxying.
+
+There exists 2 types of gateways:
+
+- Delegated: Which enables users to use any existing gateway like [Kong](https://github.com/Kong/kong).
+- Builtin: Which will configure the data plane proxy to expose external listeners to drive traffic inside the mesh.
```
Comment (michaelbeaumont) [inline docs/docs/dev/documentation/gateway.md:17]:
```suggestion
- Delegated: enables users to use any existing gateway like [Kong](https://github.com/Kong/kong).
- Builtin: configures the proxy to expose external listeners to drive traffic inside the mesh.
```

**Review by michaelbeaumont — COMMENTED** [review]
I may have missed more missing ``

File: docs/docs/dev/documentation/gateway.md [hunk 3]
```diff
@@ -1,15 +1,35 @@
...
+
+::: warning
+Gateways exist within a mesh.
+If you have multiple meshes you will want to add one gateway by mesh.
+:::
+
+## Delegated
```
[the remark this answers is not in the document]
Comment (lahabana) [reply docs/docs/dev/documentation/gateway.md:24]: Yep next release

**Review by michaelbeaumont — APPROVED** [review]
Still one unresolved conversation ^

**Pull request merged**